WordPress/Classes/Networking/WordPressClient.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+import WordPressAPI
+import Network
+
+struct WordPressSite {
+    enum SiteType {
+        case dotCom(emailAddress: String, authToken: String)
+        case selfHosted(username: String, authToken: String)
+    }
+
+    let baseUrl: String
+    let type: WordPressSite.SiteType
+
+    init(baseUrl: ParsedUrl, type: WordPressSite.SiteType) {
+        self.baseUrl = baseUrl.url()
+        self.type = type
+    }
+
+    static func from(blog: Blog) throws -> WordPressSite {
+        let url = try ParsedUrl.parse(input: blog.getUrlString())
+        if let account = blog.account {
+            return WordPressSite(baseUrl: url, type: .dotCom(
+                emailAddress: account.email,
+                authToken: account.authToken
+            ))
+        } else {
+            return WordPressSite(baseUrl: url, type: .selfHosted(
+                username: try blog.getUsername(),
+                authToken: try blog.getApplicationToken())
+            )
+        }
+    }
+}
+
+actor WordPressClient {
```

Comment:
What's the role of `WordPressClient`?

Reply:
A wrapper of the REST API client implemented in wordpress-rs. @jkmassel would be a better person to answer this in detail.

Reply:
IMHO the idea would be that it could encapsulate client access details for a specific site. It would answer questions like:

- Can we currently reach this site via the network? (if not, we should display an offline banner)
- Which radio are we using to access this site? (controls the extent to which we do preloading)
- Does the user have permission to perform `$OPERATION` on this site? (it'd cache this, but could reload the permissions on a force-refresh)

Possibly other stuff – I don't want it to become the everything object, but I also don't want to thread 10 different objects through the object hierarchy either.